File: U1/TI1.3 RESTFULL_JAVA_SINBDD_GR01/02 CLICON/ConUni_Rest_Cliente_Java_GR01/src/main/java/ec/edu/monster/models/ConversionRequest.java
package ec.edu.monster.models;

/**
 * Modelo para enviar solicitudes de conversión al servidor REST
 * Representa el JSON que se envía en el body del POST
 * 
 * Ejemplo JSON:
 * {
 *   "valor": "100",
 *   "unidadOrigen": "Metro",
 *   "unidadDestino": "Milla"
 * }
 * 
 * @author jeffe
 */
public class ConversionRequest {
    
    private String valor;
    private String unidadOrigen;
    private String unidadDestino;
    
    // Constructor vacío (requerido para Gson)
    public ConversionRequest() {
    }
    
    // Constructor con parámetros
    public ConversionRequest(String valor, String unidadOrigen, String unidadDestino) {
        this.valor = valor;
        this.unidadOrigen = unidadOrigen;
        this.unidadDestino = unidadDestino;
    }
    
    // Getters y Setters
    public String getValor() {
        return valor;
    }
    
    public void setValor(String valor) {
        this.valor = valor;
    }
    
    public String getUnidadOrigen() {
        return unidadOrigen;
    }
    
    public void setUnidadOrigen(String unidadOrigen) {
        this.unidadOrigen = unidadOrigen;
    }
    
    public String getUnidadDestino() {
        return unidadDestino;
    }
    
    public void setUnidadDestino(String unidadDestino) {
        this.unidadDestino = unidadDestino;
    }
    
    @Override
    public String toString() {
        return "ConversionRequest{" +
               "valor='" + valor + '\'' +
               ", unidadOrigen='" + unidadOrigen + '\'' +
               ", unidadDestino='" + unidadDestino + '\'' +
               '}';
    }
}

File: U1/TI1.4 RESTFULL_DOTNET_SINBDD_GR01/02CLICON/ConUni_Client_DotNet/Models/ConversionRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConUni_Client_DotNet.Models
{
    /// <summary>
    /// Modelo de solicitud para conversiones de unidades
    /// DEBE coincidir con el modelo del servidor
    /// </summary>
    public class ConversionRequest
    {
        public string Valor { get; set; } = string.Empty;
        public string UnidadOrigen { get; set; } = string.Empty;
        public string UnidadDestino { get; set; } = string.Empty;

        public ConversionRequest() { }

        public ConversionRequest(string valor, string unidadOrigen, string unidadDestino)
        {
            Valor = valor;
            UnidadOrigen = unidadOrigen;
            UnidadDestino = unidadDestino;
        }
    }
}